code/modules/events/space_ninja.dm

```diff
@@ -0,0 +1,50 @@
+/datum/round_event_control/space_ninja
+	name = "Spawn Space Ninja"
+	typepath = /datum/round_event/ghost_role/space_ninja
+	max_occurrences = 1
+	earliest_start = 20 MINUTES
+	min_players = 20
+
+/datum/round_event/ghost_role/space_ninja
+	minimum_required = 1
+	role_name = "Space Ninja"
+
+/datum/round_event/ghost_role/space_ninja/spawn_role()
+	var/list/spawn_locs = list()
+	for(var/obj/effect/landmark/carpspawn/carp_spawn in GLOB.landmarks_list)
+		if(!isturf(carp_spawn.loc))
+			stack_trace("Carp spawn found not on a turf: [carp_spawn.type] on [isnull(carp_spawn.loc) ? "null" : carp_spawn.loc.type]")
+			continue
+		spawn_locs += carp_spawn.loc
+	if(!spawn_locs.len)
+		message_admins("No valid spawn locations found, aborting...")
+		return MAP_ERROR
+
+	//selecting a candidate player
+	var/list/candidates = get_candidates(ROLE_NINJA, null, ROLE_NINJA)
```

Comment:
this is different than poll right?

Reply:
Yes, this is a specific method used for all midround antagonists.